doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -866,16 +866,17 @@ notations are given below. The optional :production:`scope` is described in
 :ref:`Scopes`.
 
 .. productionlist:: coq
-   notation      : [Local] Notation `string` := `term` [`modifiers`] [: `scope`].
-                 : [Local] Infix `string` := `qualid` [`modifiers`] [: `scope`].
-                 : [Local] Reserved Notation `string` [`modifiers`] .
+   notation      : [Local] Notation `string` := `term` [(`modifiers`)] [: `scope`].
```

Comment:
Do production lists not use the `{? ...}` syntax for optional things?

Reply:
They don't indeed. Maybe we'll change this in the future, but `productionlist` is a built-in Sphinx directive, whereas everything regarding the `{? ... }` syntax is coqrst-specific. BTW, maybe this should be upstreamed. WDYT @cpitclaudel?

Reply:
I'm not sure.  I think they'd recommend putting it in a sphinxcontrib, and then it would increae the burden for us over putting it in coqdomain, with little to gain (on the other hand, if another project wants to use it, we could package it separately for them.  But I think we should wait until someone asks).